test/upgrade_test.go

```diff
@@ -0,0 +1,180 @@
+//go:build e2e
```

Comment:
As is it will be executed on all `go_test_e2e` run which.. might or might not be something we want ?